test/e2e/allowed_source_ranges_test.go

```diff
@@ -0,0 +1,396 @@
+	ic := newLoadBalancerController(name, domain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		DNSManagementPolicy: operatorv1.ManagedLoadBalancerDNS,
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	t.Cleanup(func() { assertIngressControllerDeleted(t, kclient, ic) })
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+
+	// Set service.beta.kubernetes.io/load-balancer-source-ranges annotation and see if it will be reflected in AllowedSourceRanges in IngressController's status
+	lbService.Annotations[ingresscontroller.LoadBalancerSourceRangesAnnotation] = "127.0.0.0/8"
+	if err := kclient.Update(context.TODO(), lbService); err != nil {
+		t.Fatalf("failed to update service: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+	err := wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
+		cidrs := []string{}
+		if ic.Status.EndpointPublishingStrategy != nil {
+			lb := ic.Status.EndpointPublishingStrategy.LoadBalancer
+			if lb != nil && len(lb.AllowedSourceRanges) > 0 {
+				for _, cidr := range lb.AllowedSourceRanges {
+					cidrs = append(cidrs, string(cidr))
+				}
+			}
+		}
+
+		annotationRanges := strings.Split(lbService.Annotations[ingresscontroller.LoadBalancerSourceRangesAnnotation], ",")
+		t.Logf("Annotation: %v, AllowedSourceRanges: %v", annotationRanges, cidrs)
+		if reflect.DeepEqual(annotationRanges, cidrs) {
+			return true, nil
+		}
+
+		if err := kclient.Get(context.TODO(), name, ic); err != nil {
+			return false, fmt.Errorf("failed to get ingresscontroller: %w", err)
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("expected the annotation to be reflected in status.allowedSourceRanges: %v", err)
+	}
+
+	// Set .Spec.LoadBalancerSourceRanges and see if it will be reflected in AllowedSourceRanges in IngressController's status
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+	lbService.Spec.LoadBalancerSourceRanges = []string{"0.0.0.0/0"}
+	if err := kclient.Update(context.TODO(), lbService); err != nil {
+		t.Fatalf("failed to update service: %v", err)
+	}
+
+	if err = kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
```

Comment:
What do you think about deleting this `Get` and then moving the one inside the polling loop to the top of the polling loop?

Reply:
Makes sense. I changed its place.